### ISSUE TITLE
Headless Browsers Not Finding Splashpage

### ISSUE THREAD

**Issue by rb1997** [kdcz/3s_atc#31]
I think adidas has blocked headless browsers on the website entirely. Any way to implement chromedriver as opposed to phantom JS?

Thanks for all the hard work, much prefer this to SP.

**Comment by kdcz** [comment]
Done, download the new version!

**Comment by rb1997** [comment]
Thanks!!

One suggestion, the combined opening of many sessions of chromedriver is enough to crash my pc. Although when i open several instances of 3s_atc each with 10 processes in each rather than 30 at once, my pc is running fine at only 25% capacity. 

Any way of adding in a feature that sets up the chromedrivers more gradually?

**Comment by jalex1** [comment]
I have noticed this also^^^

**Comment by jalex1** [comment]
Also is it possible to backdoor when trying to get products with hmac?

**Comment by kdcz** [comment]
I'll check that.
@jalex1 what do you mean ?

**Comment by jalex1** [comment]
Doesn't matter, I figured out that hmac means you can't.

**Comment by rb1997** [comment]
@kdcz looking at task manager when the sessions are setting up, my disk usage % rises all the way up to 100%. Ive found if i use around 20 sessions my computer copes and the disk usage gradually stabilizes, reducing back down to a steady 20%ish however if i use more sessions it is too much for it to continue and the program crashes.

RAM and CPU are coasting using only around 30%.

**Comment by kdcz** [comment]
Sorry for the inactivity guys I had some stuff to do for school but now I'm done and I'm working on a new method without any driver because it takes too much RAM. It'll be ready for the 24th release(zebra restock).

**Comment by rb1997** [comment]
@kdcz please don't feel the need to apologise bro, school work should always come first. I'm sure im not alone when i say we are all really appreciative of all the hard work you've put into this, thanks so much man!!

Its making me feel lazy, I need to learn to code!!

**Comment by kdcz** [comment]
Thanks for the support !

**Comment by rb1997** [comment]
@kdcz 
Okay, some feedback from the yeezy powerphase drop yesterday. Bot was very unstable due to the strain of starting so many chrome processes. This meant i could only run around 3 non refresh and 10 refresh MAX in multisession mode.

1) EVERY SINGLE CHROME INSTANCE PASSED THE SPLASHPAGE! However the bot did not actually detect this and i only realised that this had happened when i manually made each task visible to discover i had actually got through.
![image](https://cloud.githubusercontent.com/assets/28559487/26783641/20165136-49f1-11e7-9963-9b7fd18b26cc.png)

2) When i made the browsers visible in an attempt to manually proceed to atc and checkout, the browsers were often unresponsive with the size selection page stating that my browser was not supported for their captcha system. This was not the case for about 60% of the windows so i believe this error COULD be to do with my VPN. 
Sometimes an error was displayed, other times no evidence of captcha showed at all:
![image](https://cloud.githubusercontent.com/assets/28559487/26783598/f4364df0-49f0-11e7-9712-a60feb45231b.png)

3) Of the tasks that passed splash AND captcha was available, due to it being on a fresh browser i was given infinite captchas, i bypassed this by logging into my gmail in a new tab and then was able to successfuly get through splash instantly.  This meant i was able to checkout 3 pairs via this method.

To summarise:

- Current setup is too dependent on cpu, ram and disk usage. Causes instability and sluggishness.

- Current version **did not detect that the splash page was passed** - potentially due to captcha glitch.

- Did aid in copping 3 pairs but the majority i had to do manually. If all errors are ironed out this has a HUGE amount of potential.

Thanks for your efforts and i hope this fairly vague feedback is somewhat useful.

**Comment by kdcz** [comment]
First I'd like to thank you for your support and activity, and your great feedback.
As I said before, I am working(it's almost done) on a new method with headless browsers so it should use much less RAM than chrome does. Secondly, it seems like the product page identifier wasn't the right one for this release and I apologize for it, as soon as possible I'll start creating a thread here for each "important" release in case you have a problem. Regarding the captcha issue, I will add a method to login to gmail so I can retrieve then google cookies and send them to all browsers to avoid such issues.Finally, I'll improve the automatic add to cart process so you won't have to cop manually.

Unfortunately I wasn't able to cop since I'm living in Europe, congratulations for your 3 Ws and thank you for the feedback !

**Comment by kdcz** [comment]
New update : 

 -We now have our own browser(3s_atc - browser) based on CEFSharp
-3s_atc and 3s_atc-browser are communicating between them to transfer all informations
-It uses less RAM than Selenium method with Chrome(1 instance of 3s_atc browser = around 25 Mo of RAM Usage)
-Fixed product page detection

**Comment by rb1997** [comment]
Very impressive work, will test this over the next few days and let you know how it goes. Thanks so much once again!
